### PR TITLE
fozzie-components@v2.5.0 & f-button@v0.1.0 – fozzie package updates and initial f-button component setup

### PR DESCRIPTION
**Pull request by ashleynolan**
### Added
- First iteration of button component. Contains variants for primary, secondary, ghost, outline and btnLink.

---

### Added
- Upgrading `fozzie-colour-palette` and `fozzie` package versions.

---

## UI Review Checks

Screen Shot 2020-10-21 at 11 01 16

- [x] README and/or UI Documentation has been created
- [ ] Unit tests have been created (will follow in separate PR)
- [x] This code has been checked with regard to [our accessibility standards](http://fozzie.just-eat.com/documentation/general/accessibility/checklist)

## Browsers Tested

- [x] Chrome (latest)

### CODE REVIEW

**Review by kevinrodrigues — DISMISSED** [review]

**Comment by kevinrodrigues** [comment]
@ashleynolan LGTM there's a conflict tho.

**Comment by ashleynolan** [comment]
@kevinrodrigues should be resolved now 👍

**Review by kevinrodrigues — DISMISSED** [review]

File: packages/f-button/src/components/Button.vue [hunk 1]
```diff
@@ -0,0 +1,340 @@
...
+
+export default {
+    name: 'VueButton',
+    components: {},
+    props: {
+        buttonType: {
+            type: String,
+            default: 'primary'
+        },
+        buttonSize: {
```
Comment (DamianMullins) [inline packages/f-button/src/components/Button.vue:24]: I'm still not sold on using string values for props, but if that's the way we want to go then I think we should add a validator here to make sure valid values are being set?

File: packages/f-button/src/components/Button.vue [hunk 2]
```diff
@@ -0,0 +1,340 @@
...
+$btn-sizeLarge-padding          : 13px 1.2em 15px;
+
+$btn-sizeSmall-padding          : 7px 1em 9px;
+
+$btn-sizeXSmall-padding         : 5px 0.5em 7px;
+$btn-sizeXSmall-lineHeight      : 1;
+
+
+.o-btn {
+    display: inline-block;                      /* [1] */
```
Comment (DamianMullins) [inline packages/f-button/src/components/Button.vue]: Do you need the comments on these lines?

File: packages/f-button/src/components/Button.vue [hunk 3]
```diff
@@ -0,0 +1,340 @@
...
+    font-family: $btn-default-font-family;      /* [3] */
+    @include font-size($btn-default-font-size); /* [3] */
+    cursor: pointer;                            /* [4] */
+    margin: 0;                                  /* [5] */
+    padding: $btn-default-padding;              /* [5, 6] */
+    overflow: visible;                          /* [7] */
+    text-align: center;
+    font-weight: $btn-default-weight;
+
+    // You may want to change this
```
Comment (DamianMullins) [inline packages/f-button/src/components/Button.vue]: Who is this comment meant for and why would they want to change the value?

File: packages/f-button/src/components/Button.vue [hunk 4]
```diff
@@ -0,0 +1,340 @@
...
+    name: 'VueButton',
+    components: {},
+    props: {
+        buttonType: {
+            type: String,
+            default: 'primary'
+        },
+        buttonSize: {
+            type: String,
+            default: 'medium'
```
Comment (jamieomaguire) [inline packages/f-button/src/components/Button.vue:26]: if there's only a select few options that the size could be i.e. xsmall, medium etc (I don't actually know whether or not this is the case!), is it worth adding any prop validation here?

File: packages/f-button/src/components/Button.vue [hunk 5]
```diff
@@ -0,0 +1,340 @@
...
+}
+
+/**
+ * ==========================================================================
+ * Btn Background Colour modifiers
+ * ==========================================================================
+ */
+
+/**
+ * Modifier – .o-btn--primary
```
Comment (DamianMullins) [inline packages/f-button/src/components/Button.vue:156]: Is the class name necessary here when it's declared a few lines down?

File: packages/f-button/src/components/Button.vue [hunk 6]
```diff
@@ -0,0 +1,340 @@
...
+.o-btn--sizeLarge {
+    @include font-size($btn-sizeLarge-font-size);
+    padding: $btn-sizeLarge-padding;
+
+    &.o-btn--primary {
+        background-color: $orange;
+    }
+}
+
+// .o-btn--sizeMedium {}
```
Comment (DamianMullins) [inline packages/f-button/src/components/Button.vue]: Can we remove this?

**Review by jamieomaguire — DISMISSED** [review]

File: packages/f-button/src/components/Button.vue (hunk 2 above)
Comment (ashleynolan) [reply packages/f-button/src/components/Button.vue]: Think this was a hangover from old button styles – I'll remove them

File: packages/f-button/src/components/Button.vue (hunk 3 above)
Comment (ashleynolan) [reply packages/f-button/src/components/Button.vue]: Again – copy/paste from `fozzie`. Don't think we need this anymore though 😆

File: packages/f-button/src/components/Button.vue (hunk 4 above)
Comment (ashleynolan) [reply packages/f-button/src/components/Button.vue:26]: Yep – I'm doing a separate PR for unit tests, so if it's ok, I'll add this to that PR?

File: packages/f-button/src/components/Button.vue (hunk 5 above)
Comment (ashleynolan) [reply packages/f-button/src/components/Button.vue:156]: I kinda like it, but I guess that's just personal preference. Happy to remove if you think it's not needed?

File: packages/f-button/src/components/Button.vue (hunk 6 above)
Comment (ashleynolan) [reply packages/f-button/src/components/Button.vue]: Will do

File: packages/f-button/src/components/Button.vue (hunk 5 above)
Comment (DamianMullins) [reply packages/f-button/src/components/Button.vue:156]: I'm happy to keep it if it's adding value, sure. Explicitly marking it as a modifier is nice.

File: packages/f-button/src/components/Button.vue [hunk 7]
```diff
@@ -0,0 +1,340 @@
...
+            if (this.buttonSize === 'xsmall') {
+                return this.buttonSize.slice(0, 2).toUpperCase() + this.buttonSize.slice(2); // xsmall -> XSmall
+            }
+            return this.buttonSize.charAt(0).toUpperCase() + this.buttonSize.slice(1); // capitalize the first letter of the prop
+        }
+    }
+};
+</script>
+
+<style lang="scss" module>
```
Comment (DamianMullins) [inline packages/f-button/src/components/Button.vue:47]: When I was testing pulling button components into another project I noticed that the generated scoped class styles couldn't easily be overridden as they were added after the styles where I was trying to override thanks to the cascade. Have you hit this issue before? Or would you have any recommendations on how to resolve that?

File: packages/f-button/src/components/Button.vue (hunk 4 above)
Comment (jamieomaguire) [reply packages/f-button/src/components/Button.vue:26]: ace!

**Review by jamieomaguire — APPROVED** [review]

**Review by DamianMullins — APPROVED** [review]

File: packages/f-button/src/components/Button.vue (hunk 7 above)
Comment (ashleynolan) [reply packages/f-button/src/components/Button.vue:47]: Probably worth catching up on hangout about this as would be good to get a bit more context. I'll drop you a msg 👍

Comment (DamianMullins) [reply packages/f-button/src/components/Button.vue:47]: Yeah sounds good, I'll wait for this component to be released to npm and then set up a demo we can look at.

**Pull request merged**